ui/src/app/pages/admin/disks-table.spec.tsx

```diff
@@ -0,0 +1,200 @@
+import '@testing-library/jest-dom';
+
+import * as fp from 'lodash/fp';
+
+import {
+  AppType,
+  Disk,
+  DiskStatus,
+  DiskType,
+  ListDisksResponse,
+} from 'generated/fetch';
+
+import { screen } from '@testing-library/dom';
+import {
+  fireEvent,
+  render,
+  waitFor,
+  waitForElementToBeRemoved,
+  within,
+} from '@testing-library/react';
+import * as swaggerClients from 'app/services/swagger-fetch-clients';
+import moment from 'moment';
+
+import { DisksTable } from './disks-table';
+
+const convertDate = (originalDate: String) => {
+  const date = originalDate.substring(0, 10);
+  const hour =
+    parseInt(originalDate.substring(11, 13), 10) + moment().utcOffset() / 60;
+  const minutes = originalDate.substring(14, 16);
+  return `${date} ${hour}:${minutes}`;
+};
+
+const getEnvironmentType = (isGceRuntime: boolean, appType: AppType) => {
+  return isGceRuntime ? 'Jupyter' : fp.capitalize(appType.toString());
+};
+
+const mockJupyterDisk: Disk = {
+  size: 1000,
+  diskType: DiskType.Standard,
+  isGceRuntime: true,
+  name: 'mock-disk1',
+  blockSize: 1,
+  status: DiskStatus.Ready,
+  appType: null,
+  creator: '"evrii@fake-research-aou.org"',
+  createdDate: '2023-05-22T18:55:10.108838Z',
+};
```

Comment:
Could you move these into `disks-api-stub.ts`? I would use these in other tests and have considered adding GKE-friendly stubs to that file before.

Reply:
I like the idea of moving these somewhere common. I moved them to disks-api-stub.ts for now, but I think that it might be beneficial to move to a model where we have a mocks directory where we can mock based on entity rather api. A conversation for another time though.